File: c++/data_src_dependent/bst_shakespeare/bst_shakespeare.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/Shakespeare.h"
#include "BSTElement.h"
#include "BST.h"

using namespace bridges;


int main(int argc, char **argv) {
	string hilite_color = "orange", 
			def_color = "green",
			end_color = "red";
		
#if TESTING
                        // command line args provide credentials and server to test on
    Bridges *bridges =  new Bridges(atoi(argv[1]), argv[2], argv[3]);
    if (argc > 4)
        bridges->setServer(argv[4]);
#else
    Bridges *bridges =  new Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
                                "YOUR_API_KEY");
#endif
								// read the earth quake  data and build the BST

	bridges->setTitle("BST Example: Shakespeare Plays, Poems, Sonnets");

	DataSource *ds = new DataSource;

    vector<Shakespeare> book_list = ds->getShakespeareData();

	BST<string, Shakespeare> *bst = new BST<string, Shakespeare> ();

	for (int k = 0; k < book_list.size(); k++) {
		Shakespeare b = book_list[k];
		bst->insert(b.getTitle(), b);
	}


					// visualize the binary search tree
	bridges->setDataStructure(bst->getRoot());
	bridges->visualize();

	return 0;
}

File: java/data_src_dependent/bst_shakespeare/bst_shakespeare.java

//
// SimpleDriver.java : Driver to illustrate importing USGS earthquake data
// 
import java.util.Scanner;
import bridges.connect.Bridges;
import bridges.base.BSTElement;
import bridges.data_src_dependent.Shakespeare;
import java.util.ArrayList;

public class bst_shakespeare {
	public static final int maxElements = 100; //number of tweets

	public static void main(String[] args) throws Exception{
	
    						// Instantiate a Bridges object 
        Bridges bridges = new Bridges (Integer.parseInt(args[0]), args[1], args[2]);
        bridges.setServer (args[3]);


		bridges.setTitle("Shakespeare Sonnets, Poems, Plays");

							// Retrieve Shakespeare Data 
		ArrayList<Shakespeare> shksp_list = 
				(ArrayList<Shakespeare>) Bridges.getShakespeareData();

		BST<String, Shakespeare>  bst = new BST<String, Shakespeare>();
		for ( int i = 0; i < shksp_list.size(); i++ ){
			bst.insert(shksp_list.get(i).getTitle(), shksp_list.get(i));
		}

							// give BRIDGES handle to tree root
		bridges.setDataStructure(bst.getTreeRoot());
							// visualize the tree
		bridges.visualize();
	}
}
